nixos/modules/services/misc/snapper.nix

```diff
@@ -131,6 +153,22 @@ in
 
         };
 
+        systemd.services.snapper-subvolume-setup = {
+          description = "Creates .snapshots subvolumes for Snapper Snapshots";
+          inherit documentation;
+          script = let
+            config = cfg: "v /.snapshots 0770 root root\n" + (
+              let
+                acl = lib.concatMapStringsSep "," (x: "${x}:r-x") ((map (u: "u:${u}") cfg.allowUsers) ++ (map (g: "g:${g}") cfg.allowGroups));
+              in
+                optionalString (acl != "") "a /.snapshots - - - - ${acl}"
+            );
+            cmd = cfg: "systemd-tmpfiles --create --root ${cfg.subvolume} ${builtins.toFile "snapper-subvolume-setup.conf" (config cfg)}";
+          in
+          lib.concatStringsSep "\n" (lib.mapAttrsToList (name: cfg: cmd cfg) cfg.configs);
+          wantedBy = [ "multi-user.target" ];
+        };
+
```

Comment:
Why not use `systemd.tmpfiles.rules` for this?

Reply:
because `v` creates a directory instead of a submodule if / is not btrfs. So it is required to the --root argument to systemd-tmpfiles.

The alternative is to hack an idempotent wrapper around btrfs submodule create.